src/main/scala/nodes/learning/KernelMatrix.scala

```diff
@@ -0,0 +1,78 @@
+package nodes.learning
+
+import breeze.linalg._
+import edu.berkeley.cs.amplab.mlmatrix.{RowPartition, NormalEquations, BlockCoordinateDescent, RowPartitionedMatrix}
+import nodes.stats.{StandardScalerModel, StandardScaler}
+import nodes.util.{VectorSplitter, Identity}
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.rdd.RDD
+import scala.collection.mutable.HashMap
+import utils.{MatrixUtils, Stats}
+import workflow.{Transformer, LabelEstimator}
+
+/* Defines a distributed KernelMatrix
+ * Note this is a lazy matrix, 
+ */
+
+trait KernelMatrix {
+
+  /* Return a n x idxs.size sub matrix (every entry in RDD is a row)
+   * @param idxs the column indexes of your kernel matrix you want
+   * */
+  def apply(idxs: Seq[Int]):RDD[DenseMatrix[Double]]
+
+  def apply(idxs1: Seq[Int], idxs2: Seq[Int]): RDD[DenseMatrix[Double]]
+
+}
+
+class BlockedKernelMatrix (kernelGen: KernelGenerator, data: RDD[DenseVector[Double]],  train: Boolean, cacheKernel: Boolean) extends KernelMatrix
+{
+  val cache = HashMap.empty[(Seq[Int], Seq[Int]),RDD[DenseMatrix[Double]]]
+
+  def apply(idxs: Seq[Int]):RDD[DenseMatrix[Double]] = {
+    val kBlock =
+    if (cache contains ((idxs, Seq.empty[Int]))) {
```

Comment:
spacing here is kind of weird, maybe indent this block? or put the `if` on the same line as `=`